lib/deep-readonly/index.ts

```diff
@@ -21,10 +21,22 @@ export type DeepReadonly<Type> = Type extends Exclude<Builtin, Error>
   ? Promise<DeepReadonly<Value>>
```

Comment:
@Somnium7 can you add the iterator as a test example in `test/deep-readonly.ts`? Leave the example, that works for you, but as minimal as possible